### PR TITLE
PT-90: add batch items mutation

### PR DESCRIPTION
**Pull request by ksavosteev**
https://virtocommerce.atlassian.net/browse/PT-51

added mutation 'addItemsCart' for batch add products
changed purchase mutation signatures: currency -> currencyCode, language -> cultureName
moved AddressType to Core project (removed from order, member, purchase)

### CODE REVIEW

File: src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs [hunk 1]
```diff
@@ -163,6 +167,33 @@ public virtual async Task<CartAggregate> AddItemAsync(NewCartItem newCartItem)
             return this;
         }
 
+        public virtual async Task<CartAggregate> AddItemsAsync(ICollection<NewCartItem> newCartItem)
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs:170]: 1. newCartItem**s**;
2. IReadOnlyCollection is better IMHO.

File: src/XPurchase/VirtoCommerce.XPurchase/Schemas/InputAddItemsType.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using GraphQL.Types;
+
+namespace VirtoCommerce.XPurchase.Schemas
+{
+    public class InputAddItemsType : InputCartBaseType
+    {
+        public InputAddItemsType()
+        {
+            Field<NonNullGraphType<ListGraphType<InputNewCartItemType>>>("AddCartItems");
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.XPurchase/Schemas/InputAddItemsType.cs:9]: Just `CartItems`.

File: src/XPurchase/VirtoCommerce.XPurchase/Schemas/InputNewCartItemType.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+using GraphQL.Types;
+
+namespace VirtoCommerce.XPurchase.Schemas
+{
+    public class InputNewCartItemType : InputObjectGraphType<NewCartItem>
+    {
+        public InputNewCartItemType()
+        {
+            Field(x => x.ProductId, false).Description("Product Id");
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.XPurchase/Schemas/InputNewCartItemType.cs:9]: Add caption `nullable:`, please.

File: src/XPurchase/VirtoCommerce.XPurchase/Schemas/PurchaseSchema.cs [hunk 4]
```diff
@@ -662,6 +662,37 @@ public void Build(ISchema schema)
...
+            /// "variables": {
+            ///    "command": {
+            ///        "storeId": "Electronics",
+            ///         "cartName": "default",
+            ///         "userId": "b57d06db-1638-4d37-9734-fd01a9bc59aa",
+            ///         "currencyCode": "USD",
+            ///         "cultureName": "en-US",
+            ///         "cartType": "cart",
+            ///         "cartId": "",
+            ///         "addCartItems": [{
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.XPurchase/Schemas/PurchaseSchema.cs:678]: Just `cartItems`.

File: src/XPurchase/VirtoCommerce.XPurchase/Commands/AddCartItemsCommandHandler.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+namespace VirtoCommerce.XPurchase.Commands
+{
+    public class AddCartItemsCommandHandler : CartCommandHandler<AddCartItemsCommand>
+    {
+        public AddCartItemsCommandHandler(ICartAggregateRepository cartRepository)
+            : base(cartRepository)
+        {
+        }
+
+        public override async Task<CartAggregate> Handle(AddCartItemsCommand request, CancellationToken cancellationToken)
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.XPurchase/Commands/AddCartItemsCommandHandler.cs:13]: Add verify test, please.

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]

**Review by AliveMen — APPROVED** [review]

**Review by Vectorfield4 — APPROVED** [review]

**Review by Vectorfield4 — APPROVED** [review]

**Pull request merged**